src/test/input_byte_size_test.cc

```diff
@@ -0,0 +1,449 @@
+  {
+    FAIL_TEST_IF_ERR(TRITONSERVER_ServerDelete(server_), "deleting server");
+  }
+
+  void SetUp() override
+  {
+    ASSERT_TRUE(server_ != nullptr) << "Server has not created";
+    // Wait until the server is both live and ready.
+    size_t health_iters = 0;
+    while (true) {
+      bool live, ready;
+      FAIL_TEST_IF_ERR(
+          TRITONSERVER_ServerIsLive(server_, &live),
+          "unable to get server liveness");
+      FAIL_TEST_IF_ERR(
+          TRITONSERVER_ServerIsReady(server_, &ready),
+          "unable to get server readiness");
+      if (live && ready) {
+        break;
+      }
+
+      if (++health_iters >= 10) {
+        FAIL() << "failed to find healthy inference server";
+      }
+
+      std::this_thread::sleep_for(std::chrono::milliseconds(500));
+    }
+
+    // Create allocator with common callback
+    FAIL_TEST_IF_ERR(
+        TRITONSERVER_ResponseAllocatorNew(
+            &allocator_, ResponseAlloc, ResponseRelease,
+            nullptr /* start_fn */),
+        "creating response allocator");
+  }
+
+  void TearDown() override
+  {
+    FAIL_TEST_IF_ERR(
+        TRITONSERVER_ResponseAllocatorDelete(allocator_),
+        "deleting response allocator");
+    FAIL_TEST_IF_ERR(
+        TRITONSERVER_InferenceRequestDelete(irequest_),
+        "deleting inference request");
+  }
+
+  static TRITONSERVER_Server* server_;
+  TRITONSERVER_ResponseAllocator* allocator_ = nullptr;
+  static constexpr char input_element_[] = "Example input string";
+  static constexpr size_t kElementSizeIndicator_ = sizeof(uint32_t);
+  static char input_data_string_[];
+  TRITONSERVER_InferenceRequest* irequest_ = nullptr;
+  std::promise<TRITONSERVER_Error*> completed_;
+};
+
+TRITONSERVER_Server* InputByteSizeTest::server_ = nullptr;
+char InputByteSizeTest::input_data_string_
+    [kElementSizeIndicator_ + std::strlen(input_element_)] = {};
+
+TEST_F(InputByteSizeTest, ValidInputByteSize)
+{
+  FAIL_TEST_IF_ERR(
+      TRITONSERVER_InferenceRequestNew(
+          &irequest_, server_, "pt_identity", -1 /* model_version */),
+      "creating inference request");
+  FAIL_TEST_IF_ERR(
+      TRITONSERVER_InferenceRequestSetReleaseCallback(
+          irequest_, InferRequestComplete, nullptr /* request_release_userp */),
+      "setting request release callback");
+
+  std::vector<int64_t> shape{1, 8};
+  std::vector<float> input_data(8, 1);
+  const auto input0_byte_size = sizeof(input_data[0]) * input_data.size();
+
+  FAIL_TEST_IF_ERR(
+      TRITONSERVER_InferenceRequestAddInput(
+          irequest_, "INPUT0", TRITONSERVER_TYPE_FP32, shape.data(),
+          shape.size()),
+      "setting input for the request");
+  FAIL_TEST_IF_ERR(
+      TRITONSERVER_InferenceRequestAppendInputData(
+          irequest_, "INPUT0", input_data.data(), input0_byte_size,
+          TRITONSERVER_MEMORY_CPU, 0),
+      "assigning INPUT data");
+
+  auto p = new std::promise<TRITONSERVER_InferenceResponse*>();
```

Comment:
You don't need to `new` it, the promise can be on stack and you pass the address of it.